components/Cta/CtaContent.tsx

```diff
@@ -16,11 +16,16 @@ export const CtaContent = ({
   srText,
   children,
 }: CtaContentProps) => {
-  const heroicon = icon || styles.ctaIconMap[variant || ''] || '';
-  const iconAnimate = styles.iconAnimation[animate];
-  const iconMarginLeft = iconPosition === 'right' && children ? styles.iconLeftMargin[heroicon] || styles.iconLeftMargin.default : '';
-  const iconMarginRight = iconPosition === 'left' && children ? styles.iconRightMargin[heroicon] || styles.iconRightMargin.default : '';
-  const iconStyle = styles.iconStyles[variant || ''];
+  const heroicon = icon || styles.ctaIconMap[variant as types.CtaIconMapType] as IconType;
+  const iconAnimate = animate ? styles.iconAnimation[animate] : '';
+
+  const iconMarginLeft = iconPosition === 'right' && children && heroicon
+    ? styles.iconLeftMargin[heroicon as types.CtaIconLeftMarginType] || styles.iconLeftMarginDefault
+    : '';
+  const iconMarginRight = iconPosition === 'left' && children && heroicon
+    ? styles.iconRightMargin[heroicon as types.CtaIconRightMarginType] || styles.iconRightMarginDefault
+    : '';
+  const iconStyle = styles.iconStyles[variant as types.CtaIconStyleType];
```

Comment:
This is a lot of typecasting. Perhaps another round of looking at this is needed?

Reply:
Yeah I tried a lot of different ways but it seems like only this method worked 😬 I'll see if there are other methods.